src/lib/ngsiNotify/senderThread.cpp

```diff
@@ -99,6 +100,7 @@ void* startSenderThread(void* p)
       else
       {
         alarmMgr.notificationError(url, "notification failure for sender-thread: " + out);
+        alarmMgr.forwardingError(url, "forwarding failure for sender-thread: " + out);
```

Comment:
This is not correct...

The sender thread is dealing with either a notification or a forwarding, but not both at the same time. So if should be something like this:

```
if (<some condition that when true means the sender thread is dealing with a notification>)
{
 alarmMgr.notificationError(url, "notification failure for sender-thread: " + out);
}
else
{
  alarmMgr.forwardingError(url, "forwarding failure for sender-thread: " + out);
}
```

The key point here is the condition to distinguish between one case or another... I don't which one should be used. Any idea?

Reply:
(The same applies to the alarm reset logic, around line 93)